public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourceField/ResourceField.tsx

```diff
@@ -37,10 +36,10 @@ const ResourceField: React.FC<ResourceFieldProps> = ({
   onQueryChange,
   setResource,
   selectableEntryTypes,
-  resourceUri,
 }) => {
   const styles = useStyles2(getStyles);
   const [pickerIsOpen, setPickerIsOpen] = useState(false);
+  const resourceUri = query.azureLogAnalytics?.resource || query.azureMonitor?.resourceUri || '';
```

Comment:
yeah I am sorry to be like this, but I think you were right the first time 😆 . Sorry totally my bad. But yeah I could see some weird situation where someone first does a logs query, they they realize "oh actually I meant metrics" and then switch back but they still have the old one, you know?M might be good to be more explicit. 

I do sort of wonder if the resource picker needs to know things about queries at all, but I think this kind of refactor can happen at another time and doesn't really have much to do with your changes.